includes/abstracts/llms-abstract-user-engagement.php

```diff
@@ -28,17 +40,17 @@ abstract class LLMS_Abstract_User_Engagement extends LLMS_Post_Model {
 	public function delete() {
 
 		/**
-		 * Action fired immediately prior to the deletion of a user's earned engagement.
+		 * Action fired immediately prior to the deletion of a user's awarded engagement.
 		 *
-		 * They dynamic portion of this hook, `{$this->model_post_type}`, refers to the engagement type,
-		 * either "achievement" or "certificate".
+		 * They dynamic portion of the hook name, {@see LLMS_Abstract_User_Engagement::$model_post_type `$this->model_post_type`},
+		 * refers to the engagement type, either "achievement" or "certificate".
 		 *
 		 * @since 3.18.0
 		 * @since [version] Migrated from LLMS_User_Certificate and LLMS_User_Achievement.
 		 *
-		 * @param LLMS_User_Certificate $certificate Certificate class object.
+		 * @param LLMS_Abstract_User_Engagement $User_Engagement Achievement or certificate class object.
 		 */
-		do_action( "llms_before_delete_{$this->model_post_type}", $this );
+		do_action( "llms_before_delete_$this->model_post_type", $this );
```

Comment:
```suggestion
		do_action( "llms_before_delete_{$this->model_post_type}", $this );
```

https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#interpolation-for-naming-dynamic-hooks

In general this PR has a lot of variable interpolation within strings that *does not* use curly braces. I think it's easier to read a variable in curly braces... Perhaps it's not hte preferred general method but this is the convention we're following, please use curly braces where possible.